repl/repl.go

```diff
@@ -177,6 +176,16 @@ func EvalStringWithOption(o Options, what string) (res string, errs []string, fo
 	return
 }
 
+func extractHistoryNumber(input string) (int, bool) {
+	if len(input) > 1 && input[0] == '!' {
+		numberPart := input[1:]
```

Comment:
Have you tried looking at strings.Cut ? And its variant CutPrefix?

https://pkg.go.dev/strings#Cut
https://pkg.go.dev/strings#CutPrefix

Reply:
well, if my prefix was not a single byte, sure but I don't think it'd help much here ? (not that it matters for this but this one will perform better, and also I don't think it's any less clear? only benefit of CutPrefix would be... to advertise it exists (which I didn't I will admit))